sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/dao/EvalHierarchyRuleSupportImpl.java

```diff
@@ -0,0 +1,291 @@
+        {
+            dao.deleteSet( rules );
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.sakaiproject.evaluation.logic.external.ExternalHierarchyRules#updateNodeRule(java.lang.Long, java.lang.String, java.lang.String, java.lang.String, java.lang.Long)
+     */
+    public void updateNodeRule( Long ruleID, String ruleText, String qualifier, String option, Long nodeID )
+    {
+        logIfDebugEnabled( "updateNodeRule( " + ruleID + ", " + ruleText + ", " + qualifier + ", " + option + ", " + nodeID + " )" );
+        if( StringUtils.isBlank( ruleText ) || StringUtils.isBlank( qualifier ) || StringUtils.isBlank( option ) )
+        {
+            throw new IllegalArgumentException( "Invalid arguments for updateNodeRule(), no arguments can be null or blank: " +
+                "ruleID=" + ruleID + ", ruleText=" + ruleText + ", qualifier=" + qualifier + ", option=" + option + ", nodeID=" + nodeID );
+        }
+
+        // Generate the final rule text from the original rule text and the qualifier
+        String finalRuleText = applyQualifierToRuleText( ruleText, qualifier );
+
+        EvalHierarchyRule rule = getByID( ruleID );
+        if( rule != null )
+        {
+            rule.setRule( finalRuleText );
+            rule.setOpt( option );
+            rule.setNodeID( nodeID );
+            dao.update( rule );
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.sakaiproject.evaluation.logic.external.ExternalHierarchyRules#getRulesByNodeID(java.lang.Long)
+     */
+    public List<HierarchyNodeRule> getRulesByNodeID( Long nodeID )
+    {
+        logIfDebugEnabled( "getRulesByNodeID( " + nodeID + " )" );
+
+        List<EvalHierarchyRule> rules = getAllByNodeID( nodeID );
+        return convertRules( rules );
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.sakaiproject.evaluation.logic.external.ExternalHierarchyRules#getRuleByID(java.lang.Long)
+     */
+    public HierarchyNodeRule getRuleByID( Long ruleID )
+    {
+        logIfDebugEnabled( "getRuleByID( " + ruleID + " )" );
+
+        EvalHierarchyRule rule = getByID( ruleID );
+        if( rule == null )
+        {
+            return null;
+        }
+        else
+        {
+            return new HierarchyNodeRule( rule );
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.sakaiproject.evaluation.logic.external.ExternalHierarchyRules#getAllRules()
+     */
+    public List<HierarchyNodeRule> getAllRules()
+    {
+        logIfDebugEnabled( "getAllRules()" );
+
+        List<EvalHierarchyRule> rules = dao.findAll( EvalHierarchyRule.class );
+        return convertRules( rules );
+    }
+
+    /**
+     * Utility method to log a message at DEBUG level, only if debug logging is enabled.
+     * 
+     * @param message the message to be logged
+     */
+    private void logIfDebugEnabled( String message )
+    {
+        if( LOG.isDebugEnabled() )
+        {
+            LOG.debug( message );
+        }
+    }
```

Comment:
Might want to double check, but I believe this method is redundant as LOG.debug returns early if LOG.isDebugEnabled() is false. We usually wrap debug statements inside "if (LOG.isDebugEnabled())" to prevent unnecessary String instantiation / concatenation in the LOG.debug parameter